fix prometheus interceptors not converting context errors to gRPC codes

vtermanis opened the pull request:
First of all - thank you for a great library!

---

* [ ] I added CHANGELOG entry for this change. (**I don't see a changelog file - what am I missing?**)
* [ ] Change is not relevant to the end user.

## Changes
fix prometheus interceptors not converting context errors to gRPC codes
- Previously `context.(Canceled|DeadlineExceeded)` would result in an "Unknown" grpc_code.

We use the interceptors to not only track failures but also identify when we've accidentally passed back non-grpc errors. With this fix the context errors are correctly mapped to their gRPS status counterparts. This behaves the same as what happens in the [golang server code](https://github.com/grpc/grpc-go/blob/fe72db9589696a625af72117a600ba191227b7c5/server.go#L1339-L1345). Something similar but old existed in your deprecated repo (https://github.com/grpc-ecosystem/go-grpc-prometheus/pull/84) but it was never released nor did it make it across, it would appear.

## Verification

Added unit test which triggers `context.Canceled` to verify it's no longer recorded as `grpc_code="Unknown"`

johanbrandhorst reviewed (changes requested):
Hi, could you make this change against v2? v1 is kinda maintenance mode.

vtermanis commented:
@johanbrandhorst - is this what you meant? (I didn't realise v2 branch was current given `main` also had activity on it)

vtermanis commented:
The unit & lint test checks don't run for some reason. Any chance one of the maintainers could take a look why that is? (I don't believe I have access to see.)

vtermanis commented:
@johanbrandhorst  - apologies for the ping but it would be great to for this to be considered.

The unit & lint test checks don't run for some reason (they do pass). Any chance you or one of the maintainers could take a look why that is? (I don't believe I have access to see/re-run the tests)

johanbrandhorst commented:
> @johanbrandhorst - apologies for the ping but it would be great to for this to be considered.
> 
> The unit & lint test checks don't run for some reason (they do pass). Any chance you or one of the maintainers could take a look why that is? (I don't believe I have access to see/re-run the tests)

Sorry to have missed this - we merged v2 into main so this should be targeted at main now. Please update again 😅.

vtermanis commented:
note: Do **not** merge - the added failing test still shows the suite as passing (which obv. is wrong). I'll try to figure out what is wrong.

vtermanis commented:
Ok, now it passes:
- Corrected labels to use `bidi_stream` (instead of `server_stream`)
- Fixed typo of `Canceled` (was `Cancelled`)

Ready for review

johanbrandhorst commented:
Thanks for your contribution!
